ext/opcache/zend_file_cache.c

```diff
@@ -1004,23 +1004,55 @@ static char *zend_file_cache_get_bin_file_path(zend_string *script_path)
 /**
  * Helper function for zend_file_cache_script_store().
  *
- * @return true on success, false on error
+ * @return true on success, false on error and errno is set to indicate the cause of the error
  */
 static bool zend_file_cache_script_write(int fd, const zend_persistent_script *script, const zend_file_cache_metainfo *info, const void *buf, const zend_string *s)
 {
+	ssize_t written;
+
 #ifdef HAVE_SYS_UIO_H
 	const struct iovec vec[] = {
 		{ .iov_base = (void *)info, .iov_len = sizeof(*info) },
 		{ .iov_base = (void *)buf, .iov_len = script->size },
 		{ .iov_base = (void *)ZSTR_VAL(s), .iov_len = info->str_size },
 	};
 
-	return writev(fd, vec, sizeof(vec) / sizeof(vec[0])) == (ssize_t)(sizeof(*info) + script->size + info->str_size);
+	written = writev(fd, vec, sizeof(vec) / sizeof(vec[0]));
+	if (EXPECTED(written == (ssize_t)(sizeof(*info) + script->size + info->str_size))) {
```

Comment:
You might be able to factorize in a common var the `sizeof(*info) + script->size + info->str_size`